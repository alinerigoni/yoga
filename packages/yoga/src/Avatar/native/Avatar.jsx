import React from 'react';
import { BuildingFilled } from '@gympass/yoga-icons';
import { string, func, number } from 'prop-types';
import Box from '../../Box';
import Image from './Image';
import Placeholder from './Placeholder';

const Avatar = ({
  src,
  fill,
  placeholder,
  width,
  height,
  borderRadius,
  type,
  defaultSource,
  elevation,
  ...otherProps
}) => (
  <Box
    bgColor="elements.selectionAndIcons"
    display="flex"
    width={width}
    height={height}
    alignItems="center"
    justifyContent="center"
    borderRadius={borderRadius}
    boxShadow="small"
    style={{ elevation: 4 }}
    {...otherProps}
  >
    {src ? (
      <Image defaultSource={defaultSource} type={type} source={src} />
    ) : (
      <Placeholder fill={fill} icon={placeholder} />
    )}
  </Box>
);

Avatar.propTypes = {
  src: string,
  placeholder: func,
  fill: string,
  width: number,
  height: number,
  borderRadius: string,
  type: string,
  defaultSource: string,
  elevation: string,
};

Avatar.defaultProps = {
  src: undefined,
  placeholder: BuildingFilled,
  fill: undefined,
  width: 48,
  height: 48,
  borderRadius: 'small',
  type: 'default',
  defaultSource: undefined,
  elevation: 'small',
};

Avatar.displayName = 'Avatar';

export default Avatar;
