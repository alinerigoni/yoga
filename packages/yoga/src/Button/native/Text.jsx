import React from 'react';
import styled, { withTheme } from 'styled-components';
import { hexToRgb } from '@gympass/yoga-common';

import withTouchable from './withTouchable';
import { Label, ButtonContainer } from './Button';

const LabelText = styled(Label)`
  ${({ color }) => `color: ${color};`}
`;

const ButtonContainerText = styled(ButtonContainer)`
  ${({
    theme: {
      yoga: {
        components: { button },
      },
    },
  }) => `
    background-color: ${button.types.text.backgroundColor};
  `}
`;

const ButtonText = ({
  children,
  disabled,
  pressed,
  small,
  secondary,
  inverted,
  icon: Icon,
  theme: {
    yoga: {
      colors,
      components: { button },
    },
  },
  ...rest
}) => {
  const state = secondary ? 'secondary' : 'primary';
  let textColor = colors[state];

  if (disabled) {
    textColor = colors.text.disabled;
  } else if (inverted) {
    textColor = colors.white;
    if (pressed) {
      textColor = hexToRgb(colors.white, 0.75);
    }
  } else if (pressed) {
    textColor = hexToRgb(colors[state], 0.75);
  }

  return (
    <ButtonContainerText
      {...rest}
      disabled={disabled}
      pressed={pressed}
      small={small}
    >
      {Icon && (
        <Icon
          width={small ? button.icon.size.small : button.icon.size.default}
          height={small ? button.icon.size.small : button.icon.size.default}
          fill={textColor}
          style={{
            marginRight: button.icon.margin.right,
          }}
        />
      )}
      <LabelText
        disabled={disabled}
        pressed={pressed}
        small={small}
        secondary={secondary}
        color={textColor}
      >
        {children}
      </LabelText>
    </ButtonContainerText>
  );
};

ButtonText.propTypes = ButtonContainer.propTypes;
ButtonText.defaultProps = ButtonContainer.defaultProps;

export default withTouchable(withTheme(ButtonText));
