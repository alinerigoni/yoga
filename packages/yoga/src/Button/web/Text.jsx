import styled from 'styled-components';
import { hexToRgb } from '@gympass/yoga-common';

import Button from './Button';

const ButtonText = styled(Button)`
  ${({
    secondary,
    inverted,
    theme: {
      yoga: {
        colors,
        components: { button },
      },
    },
  }) => {
    const state = secondary ? 'secondary' : 'primary';

    return `
      background-color: ${button.types.text.backgroundColor};
      border-color: ${button.types.text.backgroundColor};
      color: ${colors[state]};

      svg {
        fill: ${colors[state]};
      }

      &:not([disabled]):hover, &:not([disabled]):focus, &:not([disabled]):active {
        background-color: ${button.types.text.backgroundColor};
        box-shadow: none;
      }

      &:not([disabled]):hover {
        color: ${hexToRgb(colors[state], 0.5)};
        svg {
          fill: ${hexToRgb(colors[state], 0.5)};
        }
      }

      &:not([disabled]):focus, &:not([disabled]):active {
        color: ${hexToRgb(colors[state], 0.75)};
        svg {
          fill: ${hexToRgb(colors[state], 0.75)};
        }
      }

      ${
        inverted
          ? `
          color: ${colors.white};
          svg {
            fill: ${colors.white};
          }

          &:not([disabled]):hover {
            color: ${hexToRgb(colors.white, 0.5)};
            svg {
              fill: ${hexToRgb(colors.white, 0.5)};
            }
          }

          &:not([disabled]):focus, &:not([disabled]):active {
            color: ${hexToRgb(colors.white, 0.75)};
            svg {
              fill: ${hexToRgb(colors.white, 0.75)};
            }
          }
        `
          : ''
      }

      &:disabled {
        background-color: ${button.types.text.backgroundColor};
        border-color: ${button.types.text.backgroundColor};
        color: ${colors.text.disabled};
        svg {
          fill: ${colors.text.disabled};
        }
      }
    `;
  }}
`;

ButtonText.displayName = 'Button.Text';

export default ButtonText;
