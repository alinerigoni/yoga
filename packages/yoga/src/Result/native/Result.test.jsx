import React from 'react';
import { render } from '@testing-library/react-native';

import { BuildingFilled } from '@gympass/yoga-icons';
import { ThemeProvider, Result } from '../..';

import Avatar from '../../Avatar';
import Text from '../../Text';

const attendanceList = [
  {
    description: 'Attendance 01',
    icon: BuildingFilled,
  },
  {
    description: 'Attendance 02',
    icon: BuildingFilled,
  },
];

const entranceList = [
  {
    children: 'From 7:30 am to 9:00 pm',
    variant: 'stamina',
  },
];

const activitiesList = [
  {
    children: 'Yoga',
    variant: 'deep',
  },
  {
    children: 'Meditation',
    variant: 'deep',
  },
  {
    children: 'Pilates',
    variant: 'deep',
  },
  {
    children: '04',
    variant: 'deep',
  },
];

const tagsList = [
  {
    children: 'Plan x Product availability',
    variant: 'informative',
  },
  {
    children: 'Restriction',
    variant: 'attention',
  },
];

describe('<Result />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Result
          rate="5.0"
          avatar={<Avatar.Circle />}
          attendances={attendanceList}
          title="Partner Name"
          subTitle="Activity"
          divided
        >
          <Result.Details items={entranceList} dots renderItem={Text.Small} />
          <Result.Details
            items={activitiesList}
            dots
            limit={3}
            renderItem={Text.Small}
          />
          <Result.Tags items={tagsList} />
          <Result.Button>Link Button</Result.Button>
        </Result>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
