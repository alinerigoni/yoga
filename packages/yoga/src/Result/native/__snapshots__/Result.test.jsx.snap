// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    display="flex"
    divided={true}
    flexDirection="row"
    style={
      Array [
        Object {
          "borderBottomColor": "#D7D7E0",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "width": "100%",
        },
      ]
    }
  >
    <View
      alignItems="center"
      bgColor="elements.selectionAndIcons"
      borderRadius="circle"
      boxShadow="small"
      display="flex"
      height={48}
      justifyContent="center"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9898A6",
            "borderRadius": 9999,
            "display": "flex",
            "height": 48,
            "justifyContent": "center",
            "shadowColor": "rgba(152, 152, 166, 0.25)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 6,
            "width": 48,
          },
          Object {
            "elevation": 4,
          },
        ]
      }
      width={48}
    >
      <RNSVGSvgView
        bbHeight="50%"
        bbWidth="50%"
        fill="#FFFFFF"
        focusable={false}
        height="50%"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": "50%",
              "width": "50%",
            },
          ]
        }
        width="50%"
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 24,
            "marginLeft": 16,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        size="xsmall"
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 12,
              "fontWeight": "400",
              "maxWidth": 288,
            },
          ]
        }
        textAlignVertical="bottom"
      >
        <View
          style={
            Array [
              Object {
                "height": 16,
                "width": 16,
              },
            ]
          }
        >
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth="100%"
            fill="#9898A6"
            focusable={false}
            height="100%"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginTop": 3,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4288190630,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGSvgView>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 0,
                "width": 4,
              },
            ]
          }
        />
        Attendance 01
        <View
          style={
            Array [
              Object {
                "height": 0,
                "width": 4,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "height": 16,
                "width": 16,
              },
            ]
          }
        >
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth="100%"
            fill="#9898A6"
            focusable={false}
            height="100%"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginTop": 3,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4288190630,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGSvgView>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 0,
                "width": 4,
              },
            ]
          }
        />
        Attendance 02
        <View
          style={
            Array [
              Object {
                "height": 0,
                "width": 4,
              },
            ]
          }
        />
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "position": "absolute",
              "right": 0,
              "top": 4,
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={12}
          bbWidth={12}
          fill="#6B6B78"
          focusable={false}
          height={12}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 12,
                "width": 12,
              },
            ]
          }
          width={12}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4285229944,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
        <Text
          size="xsmall"
          style={
            Array [
              Object {
                "color": "#6B6B78",
                "fontFamily": "Rubik",
                "fontSize": 12,
                "fontWeight": "500",
                "lineHeight": 16,
                "marginLeft": 4,
              },
            ]
          }
          variant="deep"
        >
          5.0
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 24,
              "marginTop": 4,
            },
          ]
        }
      >
        Partner Name
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            },
          ]
        }
        variant="stamina"
      >
        Activity
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "maxWidth": 288,
                "textAlignVertical": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#231B22",
                  "fontFamily": "Rubik",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                },
              ]
            }
            variant="stamina"
          >
            From 7:30 am to 9:00 pm
          </Text>
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "maxWidth": 288,
                "textAlignVertical": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#6B6B78",
                  "fontFamily": "Rubik",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                },
              ]
            }
            variant="deep"
          >
            Yoga
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": 4,
                  "width": 4,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#6B6B78",
                  "fontFamily": "Rubik",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                  "marginLeft": 4,
                  "marginRight": 4,
                },
              ]
            }
            variant="deep"
          >
            •
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": 4,
                  "width": 4,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#6B6B78",
                  "fontFamily": "Rubik",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                },
              ]
            }
            variant="deep"
          >
            Meditation
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": 4,
                  "width": 4,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#6B6B78",
                  "fontFamily": "Rubik",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                  "marginLeft": 4,
                  "marginRight": 4,
                },
              ]
            }
            variant="deep"
          >
            •
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": 4,
                  "width": 4,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#6B6B78",
                  "fontFamily": "Rubik",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                },
              ]
            }
            variant="deep"
          >
            Pilates
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": 4,
                  "width": 4,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#6B6B78",
                  "fontFamily": "Rubik",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                  "marginLeft": 4,
                  "marginRight": 4,
                },
              ]
            }
            variant="deep"
          >
            + 
            01
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": 4,
                  "width": 4,
                },
              ]
            }
          />
        </Text>
      </View>
      <ScrollView
        horizontal={true}
        showsHorizontalScrollIndicator={false}
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 8,
            },
          ]
        }
      >
        <View
          small={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#E0DFFF",
                "borderColor": "#E0DFFF",
                "borderRadius": 4,
                "borderWidth": 1,
                "justifyContent": "center",
                "paddingBottom": 0,
                "paddingLeft": 4,
                "paddingRight": 4,
                "paddingTop": 0,
                "width": "auto",
              },
              Object {
                "marginRight": 4,
              },
            ]
          }
          variant="informative"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              inverted={false}
              light={false}
              size="medium"
              style={
                Array [
                  Object {
                    "color": "#231B22",
                    "fontFamily": "Rubik",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#231B22",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "lineHeight": 16,
                  },
                ]
              }
            >
              Plan x Product availability
            </Text>
          </View>
        </View>
        <View
          small={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FCD6C3",
                "borderColor": "#FCD6C3",
                "borderRadius": 4,
                "borderWidth": 1,
                "justifyContent": "center",
                "paddingBottom": 0,
                "paddingLeft": 4,
                "paddingRight": 4,
                "paddingTop": 0,
                "width": "auto",
              },
              Object {
                "marginRight": 4,
              },
            ]
          }
          variant="attention"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              inverted={false}
              light={false}
              size="medium"
              style={
                Array [
                  Object {
                    "color": "#231B22",
                    "fontFamily": "Rubik",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#231B22",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "lineHeight": 16,
                  },
                ]
              }
            >
              Restriction
            </Text>
          </View>
        </View>
      </ScrollView>
      <TouchableWithoutFeedback
        style={
          Array [
            Object {
              "marginTop": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            disabled={false}
            pressed={false}
            secondary={false}
            style={
              Array [
                Object {
                  "color": "#D8385E",
                  "fontFamily": "Rubik",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "marginTop": 8,
                },
              ]
            }
          >
            Link Button
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  </View>
</View>
`;
